internal/core/src/segcore/StreamReduce.cpp

```diff
@@ -0,0 +1,670 @@
+// Copyright (C) 2019-2020 Zilliz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under the License
+
+#include "StreamReduce.h"
+#include "SegmentInterface.h"
+#include "segcore/Utils.h"
+#include "Reduce.h"
+#include "segcore/pkVisitor.h"
+#include "segcore/ReduceUtils.h"
+
+namespace milvus::segcore {
+
+void
+StreamReducerHelper::FillEntryData() {
```

Comment:
Seems that we are trying to FillEntryData every time. which may takes too much time ios.

Ideally we should:

1. put segments into a group. (each group could roughly around 100GB disk usage)
2. search request by group and reduce it in one time
3. if search a lot of segment in one group, using streamReducer to help to reduce the disk usage